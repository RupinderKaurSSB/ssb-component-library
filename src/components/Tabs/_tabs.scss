.ssb-tabs {
  @include roboto;
  display: flex;
  flex-direction: row;
  font-size: 18px;
  font-weight: bold;

  .navigation-item {
    @include reset-button;
    align-items: center;
    cursor: pointer;
    display: flex;
    justify-content: center;
    padding: 20px 0;
    position: relative;
    text-align: center;
    text-decoration: none;
    transition: background .2s;

    &:not(:last-of-type) {
      margin-right: $global-padding-size * 2;
    }

    &:before {
      background: $ssb-dark-5;
      bottom: 0;
      content: '';
      display: block;
      height: 0;
      left: 0;
      position: absolute;
      transition: all .2s;
      width: 100%;
    }

    &.active {

      &:before {
        background: $ssb-green-3;
        height: 4px;
      }
    }

    &:hover {

      &:before {
        height: 4px;
      }
    }
  }
}
