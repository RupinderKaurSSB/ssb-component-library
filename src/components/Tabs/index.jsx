import React, { useState } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { ssbDark5, ssbGreen2, ssbGreen3, ssbGreen4 } from '../../style/colors';
import { roboto } from '../../style/mixins';

const Tabs = ({
	activeOnInit, items, onClick,
}) => {
	const [activeItem, changeActive] = useState(activeOnInit);

	const handleClick = e => {
		onClick(e);
		changeActive(e);
	};

	return (
		<Wrapper>
			{items.map(item => (
				<NavigationItem
					onClick={() => handleClick(item.path)}
					key={item.path}
				>
					<ButtonText>{item.title}</ButtonText>
					<ActiveIndicator active={activeItem === item.path} />
				</NavigationItem>
			))}
		</Wrapper>
	);
};

const Wrapper = styled.div`
	display: flex;
	flex-direction: row;
`;

const NavigationItem = styled.div`
	align-items: center;
	cursor: pointer;
	display: flex;
	justify-content: center;
	padding: 10px;
	position: relative;
	text-align: center;
	text-decoration: none;
	transition: background .2s;
	
	&:hover {
		background: ${ssbGreen2};
	}
	
	&:focus {
		outline: ${ssbGreen4} auto 5px;
		outline-offset: -2px;
	}
`;

const ButtonText = styled.span`
	${roboto};
	color: ${ssbDark5}
`;

const ActiveIndicator = styled.div`
	background: ${ssbGreen3};
	bottom: 0;
	height: ${props => props.active ? '5px' : 0};
	left: 0;
	position: absolute;
	transition: height .2s;
	width: 100%;
`;

Tabs.defaultProps = {
	activeOnInit: '',
	onClick: () => {},
};

Tabs.propTypes = {
	activeOnInit: PropTypes.string,
	items: PropTypes.arrayOf(PropTypes.shape({
		title: PropTypes.string,
		path: PropTypes.string,
	})),
	onClick: PropTypes.func,
};

export default Tabs;
