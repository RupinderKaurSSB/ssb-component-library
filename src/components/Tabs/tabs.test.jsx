import React from 'react';
import {shallow} from 'enzyme';
import Tabs from './index';

const items = [
	{
		title: 'Get Started',
		path: 'get-started/',
	},
	{
		title: 'Components',
		path: 'components/',
	},
	{
		title: 'Templates',
		path: 'templates/',
	},
];

describe('Tabs component', () => {
	test('Matches the snapshot', () => {
		const wrapper = shallow(<Tabs items={items} />);
		expect(wrapper).toMatchSnapshot();
	});
});
