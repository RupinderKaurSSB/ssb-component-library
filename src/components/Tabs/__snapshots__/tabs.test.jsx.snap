// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs component Matches the snapshot 1`] = `
<div
  className="ssb-tabs"
>
  <button
    className="navigation-item "
    key="get-started/"
    onClick={[Function]}
  >
    <span>
      Get Started
    </span>
  </button>
  <button
    className="navigation-item "
    key="components/"
    onClick={[Function]}
  >
    <span>
      Components
    </span>
  </button>
  <button
    className="navigation-item "
    key="templates/"
    onClick={[Function]}
  >
    <span>
      Templates
    </span>
  </button>
</div>
`;
